aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -152,6 +154,7 @@ impl VMChangeSet {
         aggregator_v1_write_set: BTreeMap<StateKey, WriteOp>,
         aggregator_v1_delta_set: BTreeMap<StateKey, DeltaOp>,
         delayed_field_change_set: BTreeMap<DelayedFieldID, DelayedChange<DelayedFieldID>>,
+        reads_needing_delayed_field_exchange: BTreeMap<StateKey, (WriteOp, Arc<MoveTypeLayout>)>,
```

Comment:
Why would a "read" operation have a "WriteOp"?

Reply:
That's what it reads - a previous WriteOp from storage. we need it to get the metadata, and be able to create Modfication vs ModificationWithMetadata WriteOp later

Reply:
Note that we have to be really careful when extracting metadata from WriteOp:

DeletionWithMetadata only contains metadata in order to deal with squashing and as such, as far as I can tell. When deletion WriteOp is applied to storage, subsequent reads return None for StateValue and hence None metadata. We are careful to do the same in metadata reads during block execution as well, i.e. if multi-versioned map contains WriteOp that is DeletionWithMetadata and we try to read latest metadata, we will read None and not Deletion's metadata (https://github.com/aptos-labs/aptos-core/blob/426a0b3e7bf47af05f1907218ff5296c18acff63/types/src/write_set.rs#L156), as should be the correct behavior - again, as long as I know - @zekun000 and @msmouse to confirm

Reply:
I am assuming that read cannot return a Deletion WriteOp, but will return that value is missing?

Reply:
The metadata only matters when we are squashing write sets and charging the write set (you refund to who's indicated in the deletion metadata, for e.g.). Reads from Move shouldn't see the metadata.

Reply:
> When deletion WriteOp is applied to storage, subsequent reads return None for StateValue and hence None metadata
yes, across transaction boundaries, the metadata on deletion should be "lost". Great observation @gelash